test/e2e/setup/setup.go

```diff
@@ -164,10 +169,33 @@ func deployOperator() error {
 	); err != nil {
 		return errors.Errorf("error building operator deployment: %s", err)
 	}
+
+	if err := wait.PollImmediate(time.Second, 30*time.Second, hasDeploymentRequiredReplicas(dep)); err != nil {
+		return errors.New("error building operator deployment: the deployment does not have the required replicas")
```

Comment:
we don't expose the underlying reason (i.e. the contents of `err` ). In go it is idiomatic to wrap errors with additional context. Something like

```golang
return errors.Errorf("error building operator deployment: the deployment does not have the required replicas: %s", err)
```

Reply:
I don't think we need to wrap the error here, the code makes no decision based on the error content -- pretty much when this error happens, the user would go and describe the object.

Reply:
I think that's a good point about the lack of decision! This message as is should be enough for what we need. 👍